src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -0,0 +1,247 @@
+    """
+    image_str = f"{registry}/{image_name}:{ver}"
+
+    # Check if the image is already present on the system
+    std_out, std_err, returncode = _run_command(["shifterimg", "images"])
+
+    images = std_out.split("\n")
+    for image in images:
+        parts = image.split()
+        if len(parts) != 6:
+            continue
+        if parts[5] == image_str:
+            print(f"Shifter image {image_name}:{ver} from registry {registry} already exists.")
+            return parts[5]
+
+    # Pull the image from the registry
+    print(f"Fetching Shifter image {image_str}...")
+    std_out, std_err, returncode = _run_command(["shifterimg", "pull", image_str])
+
+    if 'FAILURE' in std_out:
+        raise ValueError(f"Error fetching Shifter image {image_str}.")
+
+    return image_str
+
+
+def _create_shifter_wrapper(job_dir, image_str):
+    """
+    Creates the Shifter wrapper script.
+    """
+
+    # The content of the Shifter wrapper script
+    shifter_wrapper = "#!/bin/bash\n\n"
+    shifter_wrapper += f"image={image_str}\n\n"
+    shifter_wrapper += "if [ $# -lt 1 ]; then\n"
+    shifter_wrapper += '    echo "Error: Missing command argument."\n'
+    shifter_wrapper += '    echo "Usage: shifter_wrapper.sh your-command-arguments"\n'
+    shifter_wrapper += "    exit 1\n"
+    shifter_wrapper += "fi\n\n"
+    shifter_wrapper += "command=\"$@\"\n\n"
+    shifter_wrapper += "echo \"Running shifter --image=$image $command\"\n\n"
+    shifter_wrapper += f"cd $HOME\n"
+    shifter_wrapper += "shifter --image=$image $command\n"  # Run the command in the Shifter environment
+
+    wrapper_file = os.path.join(job_dir, 'shifter_wrapper.sh')
+    with open(wrapper_file, "w") as f:
+        f.write(shifter_wrapper)
+
+    os.chmod(wrapper_file, 0o777)
+
+    return wrapper_file
+
+
+def _create_genome_id_file(genome_ids, genome_id_file):
+    """
+    Create a tab-separated values (TSV) file with a list of genome IDs.
+    """
+    with open(genome_id_file, 'w') as f:
+        f.write("genome_id\n")
+        for genome_id in genome_ids:
+            f.write(f"{genome_id}\n")
+
+
+def _create_task_list(source_data_dir, kbase_collection, load_ver, tool, wrapper_file, job_dir, root_dir):
+    """
+    Create task list file (tasks.txt)
+    """
+    genome_ids = [path for path in os.listdir(source_data_dir) if
+                  os.path.isdir(os.path.join(source_data_dir, path))]
+
+    genome_ids_chunks = [genome_ids[i: i + CHUNK_SIZE] for i in range(0, len(genome_ids), CHUNK_SIZE)]
+
+    task_list = '#!/usr/bin/env bash\n'
+    for idx, genome_ids_chunk in enumerate(genome_ids_chunks):
+
+        task_list += wrapper_file
+
+        genome_id_file = os.path.join(job_dir, f'genome_id_{idx}.tsv')
+        _create_genome_id_file(genome_ids_chunk, genome_id_file)
+
+        if tool == 'checkm2':
+            task_list += f'	--volume={CHECKM2_DB}:/CheckM2_database --volume={root_dir}:/checkm2_root_dir ' \
```

Comment:
What is `checkm2_root_dir`?

Reply:
removed it. something I was testing but didn't work.

Reply:
:+1: